### DISCUSSION TITLE
Is there any way to access the 2023 pre-season LT test data in Bahrain??

### DISCUSSION THREAD

**Discussion by McHouse** [theOehrly/Fast-F1#315] [Q&A] [answered]
With the GP sessions there is no problem, but with the data from the pre-season tests? Is there access?

**Comment by McHouse** [comment, marked as answer]
Doing homework, and doing some research...

ff1.Cache.enable_cache("......")
schedule = ff1.get_event_schedule(2023)
DSchedule = pd.DataFrame(schedule)


''' # Session=ff1.get_testing_session ( Year , TestDay , "Practice 1" )'''
Session=ff1.get_testing_session ( 2023 , 1 , 1 )

Session.load(laps=True, telemetry=True, weather=True, messages=True, livedata=None)
Laps=Session.laps

This seems to work.
To anyone who has been with fastf1 and python for more than a few months, is that correct?
I ask because I still have mess with the methods of version 2.3.0

**Reply by smartbackwards** [reply]
This still seems to return everything empty, just like if you were to get a race that hasn't happened yet f.e. ff1.get_session(2023,10,1), at least for me

**Reply by McHouse** [reply]
That's why I said "it seems to work".
But I expected that there would be two sessions per test day, one in the morning and one in the afternoon.
Apparently it's just one... I don't know...
Let's wait for it to finish and we'll see what it downloads.

**Reply by theOehrly** [reply]
It is only one session for the whole day and data is available now.

**Reply by McHouse** [reply]
True, but we have to prepare "the tools" for what is interesting, it is yet to come.
Oh! and the method works.
Download the two sessions concatenated, easily distinguishable by the time.

**Reply by Iskathor** [reply]
I am working with this and I get the data from these sessions

year = 2023
test_number = 1
session_number = '2'

session = fastf1.get_testing_session(year, test_number, session_number)
weekend = session.event
session.load()

**Comment by Iskathor** [comment]
I am working with this and I get the data from these sessions

year = 2023
test_number = 1
session_number = '2'

session = fastf1.get_testing_session(year, test_number, session_number)
weekend = session.event
session.load()
